tests/end_to_end/test_kakarot.py

```diff
@@ -153,20 +153,80 @@ async def test_should_return_same_as_deployed_address(
     class TestDeployExternallyOwnedAccount:
         async def test_should_deploy_starknet_contract_at_corresponding_address(
             self,
+            starknet: FullNodeClient,
             fund_starknet_address,
             deploy_externally_owned_account,
             compute_starknet_address,
             get_contract,
         ):
-            evm_address = generate_random_evm_address()
+            evm_address = generate_random_evm_address(random.randint(0, 0x5EED))
             starknet_address = await compute_starknet_address(evm_address)
-            await fund_starknet_address(starknet_address, PRE_FUND_AMOUNT / 1e18)
 
             await deploy_externally_owned_account(evm_address)
             eoa = get_contract("account_contract", address=starknet_address)
             actual_evm_address = (await eoa.functions["get_evm_address"].call()).address
             assert actual_evm_address == int(evm_address, 16)
 
+    class TestRegisterAccount:
+        async def test_should_register_account(
+            self,
+            starknet: FullNodeClient,
+            fund_starknet_address,
+            register_account,
+            compute_starknet_address,
+            get_contract,
+        ):
+            evm_address = generate_random_evm_address(random.randint(0, 0x5EED))
+            starknet_address = await compute_starknet_address(evm_address)
+            await fund_starknet_address(starknet_address, PRE_FUND_AMOUNT / 1e18)
+
+            # TODO: change the sender's address to be the account itself
```

Comment:
@ClementWalter not sure how I can get this to pass easily. do you think the unit test is sufficient?

Reply:
not sure how to do this because the passing case is a starknet account calling the function to register its corresponding EVM account, but the only way to have this mapping correct is to start from the EVM address, and deploy the corresponding starknet account, and registration is done during deployment...

consequently I think that in terms of end-to-end test, this is covered elsewhere, and unit test is actually enough